code/modules/events/immovable_rod.dm

```diff
@@ -77,7 +77,8 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 	. = ..()
 	SSaugury.register_doom(src, 2000)
 
-	destination = end
+	var/turf/real_destination = isturf(end) ? end : get_turf(end)
```

Comment:
```suggestion
	var/turf/real_destination = get_turf(end)
```
`get_turf(turf)` will just yield the same turf. Bit less of boilerplate? Just a suggestion, you can keep it your way if you prefer it.